neuro_san/service/mcp/processors/mcp_tools_processor.py

```diff
@@ -0,0 +1,155 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+"""
+See class comment for details
+"""
+from typing import Any
+from typing import Dict
+from typing import List
+
+from neuro_san.interfaces.concierge_session import ConciergeSession
+from neuro_san.internals.network_providers.agent_network_storage import AgentNetworkStorage
+from neuro_san.service.http.interfaces.agent_authorizer import AgentAuthorizer
+from neuro_san.service.generic.async_agent_service import AsyncAgentService
+from neuro_san.service.generic.async_agent_service_provider import AsyncAgentServiceProvider
+from neuro_san.session.direct_concierge_session import DirectConciergeSession
+from neuro_san.service.mcp.util.mcp_errors_util import McpErrorsUtil
+from neuro_san.service.mcp.util.requests_util import RequestsUtil
+from neuro_san.service.http.logging.http_logger import HttpLogger
+
+
+class McpToolsProcessor:
+    """
+    Class implementing "tools"-related MCP requests.
+    Overall MCP documentation can be found here:
+    https://modelcontextprotocol.io/specification/2025-06-18/server/tools
+    """
+
+    def __init__(self,
+                 logger: HttpLogger,
+                 network_storage_dict: AgentNetworkStorage,
+                 agent_policy: AgentAuthorizer):
+        self.logger: HttpLogger = logger
+        self.network_storage_dict: AgentNetworkStorage = network_storage_dict
+        self.agent_policy: AgentAuthorizer = agent_policy
+
+    async def list_tools(self, request_id, metadata: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        List available MCP tools.
+        :param request_id: MCP request id;
+        :param metadata: http-level request metadata;
+        :return: json dictionary with tools list in MCP format
+        """
```

Comment:
If you took the aforementioned factory approach, ListTools would likely be in a separate class from CallTool